jujugui/static/gui/src/app/models/models.js

```diff
@@ -1012,6 +1012,25 @@ YUI.add('juju-models', function(Y) {
     },
 
     /**
+      Get a unit by displayName.
+
+      @method getByDisplayName
+      @param {String} displayName The display name for a unit.
+      @return {Object} The matching unit with the matching display name.
+    */
+    getByDisplayName: function(displayName) {
+      var units = this.filter(function(unit) {
+        if (unit.displayName === displayName) {
+          return unit;
+        }
+      });
+      if (units.length > 0) {
```

Comment:
curious why there would be more than one unit which matches the display name? Could you add a comment to this effect.